extensions/amp-access-poool/validator-amp-access-poool.protoascii

```diff
@@ -26,3 +26,15 @@ tags: {  # amp-access-poool
   }
   attr_lists: "common-extension-attrs"
 }
+tags: {  # amp-access-poool > section
+  html_format: AMP
+  tag_name: "SECTION"
+  spec_name: "amp-access-poool > section"
```

Comment:
This syntax suggests that this is a `<section>` tag that is a child of an `<amp-access-poool>` tag, which it is not.

`<section>` is also a standard HTML tag [link](https://www.w3schools.com/tags/tag_section.asp) that does not require `amp-access-poool` to be used in AMP pages.

I'm not sure exactly what you are trying to do here, so I'm not sure I can offer up concrete suggestions. Can you explain what you'd like the validator rules to convey?

Reply:
> This syntax suggests that this is a `<section>` tag that is a child of an `<amp-access-poool>` tag, which it is not.

We used this syntax to avoid defining our validator rule on all `section` tags globally, but that was probably not the proper way to do it.

> `<section>` is also a standard HTML tag link that does not require amp-access-poool to be used in AMP pages.

Yep, I know HTML too. Same as above, we wanted to restrict the definition of our rule to `<section>` tags used alongside `amp-access-poool` extension, but again, might not be the right way.

The docs were not really clear about how we could allow a custom attribute (like `poool-access-preview` to be used on any HTML tag, so we chose `<section>`.

Do you have any suggestion on how we could achieve that particular behavior?

Reply:
> > `<section>` is also a standard HTML tag link that does not require amp-access-poool to be used in AMP pages.
> 
> Yep, I know HTML too. 

Apologies if the tone of my comment came across that way, that was not intended.

> Same as above, we wanted to restrict the definition of our rule to `<section>` tags used alongside `amp-access-poool` extension, but again, might not be the right way.
> 
> The docs were not really clear about how we could allow a custom attribute (like `poool-access-preview` to be used on any HTML tag, so we chose `<section>`.
> 
> Do you have any suggestion on how we could achieve that particular behavior?

I think I understand. This is a little outside the typical use case which is why the documentation is sparse. Here's what I suggest:

You can see an example of a per-attribute extension requirement here:
https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii#L4631 Use the same approach for your attributes.

You want to add the attribute to all existing tagspecs for the `section` tag in [`validator-main`](https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii). In order to reduce the verbosity a little, you can use an attr_list like:

```
attr_lists: {
  name="poool-access-attrs"
  attrs: {
    name: "poool-access-preview"
    requires_extension: "amp-access-poool"
  }
  attrs: {
    name: "poool-access-content"
    requires_extension: "amp-access-poool"
  }
}
```

This can then be applied to every `section` tagspec by adding:

```
  attr_lists: "poool-access-attrs"
```

To each of those tagspecs. I've not actually tried this out in a client, but it should work if I correctly understand your intention.

Reply:
> Apologies if the tone of my comment came across that way, that was not intended.

No worries ❤️ 

>  think I understand. This is a little outside the typical use case which is why the documentation is sparse. Here's what I suggest:

> You can see an example of a per-attribute extension requirement here:
https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii#L4631 Use the same approach for your attributes.

That should be fixed with the latest commit!